src/ServiceControl/HeartbeatMonitoring/HeartbeatsMonitor.cs

```diff
@@ -147,5 +149,114 @@ void Initialise()
                 }
             }
         }
+
+        class HeartbeatsWriter : FeatureStartupTask
+        {
+            private readonly IDocumentStore store;
+            public IBus Bus { get; set; }
+
+            public HeartbeatStatusProvider HeartbeatStatusProvider { get; set; }
+
+            public HeartbeatsWriter(IDocumentStore store)
+            {
+                this.store = store;
+            }
+
+            protected override void OnStart()
+            {
+                timer = new Timer(Refresh, null, TimeSpan.FromSeconds(15), Timeout.InfiniteTimeSpan);
+            }
+
+            protected override void OnStop()
+            {
+                using (var manualResetEvent = new ManualResetEvent(false))
+                {
+                    timer.Dispose(manualResetEvent);
+
+                    manualResetEvent.WaitOne();
+                }
+            }
+
+            void Refresh(object _)
+            {
+                Sync();
+
+                try
+                {
+                    timer.Change(TimeSpan.FromMinutes(2), Timeout.InfiniteTimeSpan);
```

Comment:
Does this mean when an endpoint starts beating again that we won't notify anyone for 2 minutes?